<?xml version="1.0" encoding="UTF-8"?>
<launch>
    <arg name="serial_number" default="0" />
    <!-- ROS URDF description of the ZED -->
    <arg name="camera_name" default="zed2" />
    <group>
        <param name="$(arg camera_name)_description"
            command="$(find xacro)/xacro '$(find zed_wrapper)/urdf/zed_descr.urdf.xacro'
                        camera_name:=$(arg camera_name)
                        camera_model:=zed2
                        base_frame:=camera_link
                        cam_pos_x:=0
                        cam_pos_y:=0
                        cam_pos_z:=0
                        cam_roll:=0
                        cam_pitch:=0
                        cam_yaw:=0"
        />

        <node name="$(arg camera_name)_state_publisher" pkg="robot_state_publisher"
            type="robot_state_publisher" output="screen" required="false">
            <remap from="robot_description" to="$(arg camera_name)_description" />
            <remap from="joint_states" to="zed_joint_states" />
        </node>
    </group>

    <group ns="tj2_zed">
        <node name="zed_request_camera" pkg="tj2_camera" type="zed_request_camera" output="screen"
            required="false" respawn="true">
            <param name="camera_name" value="$(arg camera_name)" />
            <param name="serial_number" value="$(arg serial_number)" />
            <remap from="color/image" to="rgb/image_rect_color" />
            <remap from="color/camera_info" to="rgb/camera_info" />
            <remap from="depth/image" to="depth/depth_registered" />
            <remap from="depth/camera_info" to="depth/camera_info" />
        </node>
    </group>
</launch>